Присобачивание бд

ph1l74 opened the pull request:
При запуске должен чекать в цикле приходящие сообщения, сохранять в базу данных разобранные чеки.

Commit:
- п
+ CR

Commit:
added last mail func
added bytesIO instead of fopen

b0g3r commented on mail.py:
```diff
@@ -1,29 +1,63 @@
 import email
-import os
-п
+from io import BytesIO as bio
```
`bio` - bad, as не нужен здесь

b0g3r commented on mail.py:
```diff
@@ -1,29 +1,63 @@
+    files = []
+
+    # session start
+    imap_session = imaplib.IMAP4_SSL(imap_server)
```
дублирование кода, нужно чет придумать с этим

b0g3r commented on main.py:
```diff
@@ -0,0 +1,11 @@
+
+print("Files in letter: ")
+for file in files:
+    print("    {}".format(file[0]))
```
эт че за таб в начале)

Commit:
config remake:
classes replaced to nametuples
 added smtp config

Commit:
too much

b0g3r commented on config.py:
```diff
@@ -1,5 +1,15 @@
 
-user = os.environ['USER']
-password = os.environ['PASSWORD']
+_Imap = collections.namedtuple('_Imap', ['server', 'user', 'password'])
```
большими

b0g3r commented on config.py:
```diff
@@ -1,5 +1,15 @@
-user = os.environ['USER']
-password = os.environ['PASSWORD']
+_Imap = collections.namedtuple('_Imap', ['server', 'user', 'password'])
+_Smtp = collections.namedtuple('_Smtp', ['server', 'port', 'user', 'password', 'sender'])
```
большими

b0g3r commented on config.py:
```diff
@@ -1,5 +1,15 @@
+_Imap = collections.namedtuple('_Imap', ['server', 'user', 'password'])
+_Smtp = collections.namedtuple('_Smtp', ['server', 'port', 'user', 'password', 'sender'])
 
+imap = _Imap(server=os.environ['IMAP_HOST'],
```
большими по желанию

b0g3r commented on config.py:
```diff
@@ -1,5 +1,15 @@
+             user=os.environ['IMAP_USER'],
+             password=os.environ['IMAP_PASSWORD'])
+
+smtp = _Smtp(server=os.environ['SMTP_HOST'],
```
большими по желанию

b0g3r commented on mail.py:
```diff
@@ -1,29 +1,84 @@
+    # close SMTP-connection
+    mail.quit()
+
+    return True
```
а когда оно не вернет тру?)

b0g3r commented on mail.py:
```diff
@@ -1,29 +1,84 @@
+    mail.sendmail(smtp_sender, smtp_recievers, mail_message)
+
+    # close SMTP-connection
+    mail.quit()
```
ну тут точно контекст-менеджер нужен

b0g3r commented on main.py:
```diff
@@ -0,0 +1,11 @@
+import config
+import mail
+
+reciever = ['astakhovfilat@gmail.com']
```
recievers*

b0g3r commented on mail.py:
```diff
@@ -1,29 +1,84 @@
 import email
-import os
-п
+from io import BytesIO as bio
```
всё ещё `as bio` - зачем?

Commit:
one more function

Commit:
miscs

b0g3r commented on config.py:
```diff
@@ -1,5 +1,15 @@
 
-user = os.environ['USER']
-password = os.environ['PASSWORD']
+_IMAP = collections.namedtuple('_Imap', ['server', 'user', 'password'])
```
первый аргумент

b0g3r commented on config.py:
```diff
@@ -1,5 +1,15 @@
-user = os.environ['USER']
-password = os.environ['PASSWORD']
+_IMAP = collections.namedtuple('_Imap', ['server', 'user', 'password'])
+_SMTP = collections.namedtuple('_Smtp', ['server', 'port', 'user', 'password', 'sender'])
```
и тут

b0g3r commented on mail.py:
```diff
@@ -1,29 +1,85 @@
+            "not found": "В письме, которое Вы отправили нет прикрелпенного изображения"}
+
+
+def get_mail_files(message_parts, files):
```
ты зачем files передаешь О_О

b0g3r commented on mail.py:
```diff
@@ -1,29 +1,85 @@
-                file_handler = open(file_path, 'wb')
-                file_handler.write(email_part.get_payload(decode=True))
-                file_handler.close()
+    if mode == 'ALL':
```
это че такое

ph1l74 replied:
Это ужас. Исправлено.

b0g3r commented on mail.py:
```diff
@@ -1,29 +1,85 @@
+    return files
+
+
+def send_mail(smtp_host, smtp_port, smtp_user, smtp_password,  smtp_sender, smtp_recievers, mail_message):
```
если эта функция не возвращает True, то в main.py if всегда будет возвращать "not send"

ph1l74 replied:
Предложения?
Дописать функцию посылки?

b0g3r replied:
либо убрать if в main'е
либо ловить здесь эксепшены и возвращать фолс если они есть (плохой вариант)

b0g3r replied:
ну и не очень понятно зачем сюда передаются значения из конфига, пусть сюда конфиг импортируется и берется из него, чем 10 аргументов

Commit:
исправление косяков

Commit:
it's alive

Commit:
it's alive

Commit:
it's alive

b0g3r commented on main.py:
```diff
@@ -0,0 +1,7 @@
+recievers = ['astakhovfilat@gmail.com']
+
+mail.send_mail(config.smtp.server, config.smtp.port, config.smtp.user, config.smtp.password,
+                  config.smtp.sender, recievers, mail.messages['not found'])
```
`mail.messages['not found']` - это эпично) оно ж там есть, зачем ты это отсюда туда передаешь)

Commit:
it's alive

b0g3r commented on db.py:
```diff
@@ -0,0 +1,24 @@
+    date_payed = Column(DateTime)
+    date_added = Column(DateTime, default=func.now())
+
+def reciept_add(id, number, date_payed, date_added):
```
это должен быть метод в Receipt

b0g3r commented on ocr.py:
```diff
@@ -0,0 +1,13 @@
+
+def image_to_text(image):
+    image_data = image[0][1].read()
+    image_object = Image.open(BytesIO(image_data))
```
image[0][1] - уже BytesIO
ты перегоняешь из пустого в порожнее:
image[0][1] - BytesIO
image[0][1].read() - bytes
BytesIO(image[0][1].read()) - BytesIO

b0g3r commented on ocr.py:
```diff
@@ -0,0 +1,13 @@
+print(tesserocr.get_languages())
+
+
+def image_to_text(image):
```
написано image а на вход прилетает images

b0g3r commented on main.py:
```diff
@@ -0,0 +1,14 @@
+# mail.send_mail(config.smtp.server, config.smtp.port, config.smtp.user, config.smtp.password,
+#                   config.smtp.sender, recievers, mail.messages['not found'])
+
+image = mail.get_last_attachment(config.imap.server, config.imap.user, config.imap.password)
```
написано attachment, а возвращает attachments

b0g3r commented on db.py:
```diff
@@ -0,0 +1,24 @@
+class Reciept(Base):
+    __tablename__ = "reciept"
+    id = Column(Integer, primary_key=True)
+    number = Column(String)
```
все ещё не нравится

b0g3r commented on db.py:
```diff
@@ -0,0 +1,24 @@
+engine = create_engine('sqlite:///db.sqlite')
+session = sessionmaker()
+session.configure(bind=engine)
+Base.metadata.create_all(engine)
```
у тебя база создается при каждом импорте db

ph1l74 replied:
Я не знаю, что с этим делать :/

b0g3r replied:
вынести в отдельный модуль (аля init.py)
обернуть в if main

Commit:
corrections:
db.py: func in class
mail.py: one more func + bugfixes
main.py: main func
ocr.py: - bytesIO

Commit:
it's alive

b0g3r commented on main.py:
```diff
@@ -0,0 +1,22 @@
+import ocr
+import db
+
+def Check():
```
тебе пайчарм не кричит что ты функцию с большой назвал?)

Commit:
sync

Commit:
mail.py works with batch of files
main func in main totally reworked

Commit:
mail.py works with batch of files
main func in main totally reworked

b0g3r commented on mail.py:
```diff
@@ -1,29 +1,101 @@
+    # take in bytes
+    email_text = email.message_from_bytes(email_body)
+
+    # write all bytes in list
```
ты записываешь не байты в лист, а file-like объекты

b0g3r commented on mail.py:
```diff
@@ -1,29 +1,101 @@
+
+def get_mail_sender(message_parts):
+
+    re_mask = re.compile(r'[a-zA-Z0-9._-]*@[a-zA-Z0-9._-]*')
```
1. re.compile внутри блока кода, в котором он используется не имеет смысла: compile используется для часто-используемых регулярок и должен делаться, например, один раз на модуль, т.е. есть смысл поднять его куда-нибудь сильно выше в коде.
2. проверять емейлы регуляркой не нужно. Вот как на самом деле выглядит правильная регулярка для емейлов: http://www.ex-parrot.com/~pdw/Mail-RFC822-Address.html. Твоя регулярка покрывает 99% случаев, но нет, спецификация емейла позволяет творить лютейшую херню
Лучше научиться как-нибудь по-другому вытаскивать мыло, без регулярки.

b0g3r commented on mail.py:
```diff
@@ -1,29 +1,101 @@
+    return email_sender
+
+
+def get_attachments(imap_object, count=1):
```
опять зачем-то передается конфиг. пусть в этом модуле он импортится и используется, а не передается в функцию. Нужна ли тебе такая сильная модульность и переиспользование?

b0g3r commented on mail.py:
```diff
@@ -1,29 +1,101 @@
+    mail.send_message(message)
+    mail.quit()
+
+    return "Mail sent"
```
Зачем это?

b0g3r commented on main.py:
```diff
@@ -0,0 +1,37 @@
+
+
+def check(count=1):
+
```
не хватает докстринга или нормального названия функции. пока что это функция, которая проверяет один раз :)

b0g3r commented on main.py:
```diff
@@ -0,0 +1,37 @@
+    # get attachments from number of files
+    attachments = mail.get_attachments(config.imap, count)
+    recognized_text = []
+    if attachments:
```
не нужен, если attachments - пустой список, то for просто не выполнится ни разу

b0g3r commented:
Сделал ревью, но хотелось бы напомнить о сути и цели PR. Давай либо менять цель PR и ставить какую-нибудь другую, либо возвращаться к этой и делать её. Скоро код слишком разрастется и мне будет сложно его ревьюить
